docs/google/best_practices.md

```diff
@@ -58,7 +74,69 @@ using a error rate of 0.5% per gate, a circuit of depth 20 and width 20 could be
 at 0.995^(20 * 20) = 0.135. Using separate error rates per gates (i.e. based on calibration
 metrics) or a more complicated noise model can result in more accurate error estimation.
 
-## Use sweeps when possible
+### Terminal Measurements
+
+Make sure that measurements are kept in the same moment as the final moment in
+the circuit.  Make sure that any circuit optimizers do not alter this by
+incorrectly pushing measurements forward. This behavior can be avoided by
+measuring all qubits with a single gate or by adding
+the measurement gate after all optimizers have run.
+
+Currently, only terminal measurements are supported by the hardware.  If you
+absolutely need intermediate measurements for your application, reach out to
+your Google sponsor to see if they can help devise a proper circuit using
+intermediate measurements.
+
+
+### Keep qubits busy
+
+Qubits that remain idle for long periods tend to dephase and decohere. Inserting a
+[Spin Echo](https://en.wikipedia.org/wiki/Spin_echo) into your circuit onto
+qubits that have long idle periods, such as a pair
+of involutions, such as two successive Pauli Y gates, will generally increase
+performance of the circuit.
+
+### Delay initialization of qubits
+
+The |0⟩ state is more robust than the |1⟩ state. As a result, one should
+not initialize a qubit to |1⟩ at the beginning of the circuit until shortly
+before other gates are applied to it.
+
+### Align single-qubit and two-qubit layers
+
+Devices are generally calibrated to circuits that alternate single-qubit gates with
+two-qubit gates in each layer. Staying close to this paradigm will often improve
+performance of circuits.  This will also reduce the circuit's total duration,
+since the duration of a moment is its longest gate.  Making sure that each layer
+contains similar gates of the same duration can be challenging, but it will
+likely have a measurable impact on the fidelity of your circuit.
+
+Devices generally operate in the Z basis, so that rotations around the Z axis will become
+book-keeping measures rather than physical operations on the device. The EjectZ optimizer
+included in optimizer lists for each device will generally compile these operations out
+of the circuit by pushing them back to the next non-commuting operator. If the resulting
+circuit still contains Z operations, they should be aggregated into their own moment,
+if possible.
```

Comment:
If you have extra moments because of the Z gates, it can lengthen the duration of the circuit.  I tried to clarify my comments.  See if you like the new wording better.

Like, if you have
```
---Z---X---
---X-------
```